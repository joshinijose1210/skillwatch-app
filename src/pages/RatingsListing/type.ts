export interface RatingsData {
    checkInRating: number;
    employeeId: string;
    firstName: string;
    lastName: string;
    reviewCycleId: number;
}

export interface RatingsListing {
    ratingListing: RatingsData[];
    ratingListingCount: number;
}

export interface ReviewCycleOptions {
    label: string;
    value: string | number;
}

export interface LocationState {
    title: string;
    reviewCycle: string | number;
    ratingLabel: string;
}
