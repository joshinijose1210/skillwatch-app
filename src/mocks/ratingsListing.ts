const BASE_URL = 'https://dummy-api.example.com';
export const RatingsListingMockData = {
    get: {
        ratingListing: {
            url: `${BASE_URL}/api/analytics/rating-listing/?organisationId=1&reviewCycleId=150&ratingType=meetsExpectations&employeeId=-99&page=undefined&limit=10`,
            urlWithPage: `${BASE_URL}/api/analytics/rating-listing/?organisationId=1&reviewCycleId=150&ratingType=meetsExpectations&employeeId=-99&page=1&limit=10`,
            responseCode: 200,
            data: {
                ratingListingCount: 11,
                ratingListing: Array.from({ length: 11 }).map((_, i) => ({
                    reviewCycleId: 150,
                    id: 110 + i,
                    employeeId: `SR00${i + 1}`,
                    firstName: 'Moly',
                    lastName: 'Agarwal',
                    checkInRating: 3.0
                }))
            }
        },
        reviewCycle: {
            url: `${BASE_URL}/api/review-cycle/getAll?organisationId=1`,
            responseCode: 200,
            data: {
                totalReviewCycles: 1,
                reviewCycles: [
                    {
                        organisationId: 1,
                        reviewCycleId: 150,
                        startDate: 1698796800000,
                        endDate: 1701302400000,
                        publish: true,
                        lastModified: 1699718185091,
                        selfReviewStartDate: 1699574400000,
                        selfReviewEndDate: 1700438400000,
                        managerReviewStartDate: 1700006400000,
                        managerReviewEndDate: 1700870400000,
                        checkInWithManagerStartDate: 1700956800000,
                        checkInWithManagerEndDate: 1701302400000
                    }
                ]
            }
        },
        employeesData: {
            url: `${BASE_URL}/api/super-admin/employees/?teamId=-99&designationId=-99&organisationId=1&roleId=-99&searchText=&departmentId=-99`,
            responseCode: 200,
            data: {
                totalEmployees: 11,
                employees: Array.from({ length: 11 }).map((_, i) => ({
                    organisationId: 1,
                    id: 100 + i,
                    employeeId: `SR00${i + 1}`,
                    firstName: 'Moly',
                    lastName: 'Agarwal',
                    status: true
                }))
            }
        },
        searchEmployeeData: {
            url: `${BASE_URL}/api/analytics/rating-listing/?organisationId=1&reviewCycleId=150&ratingType=meetsExpectations&employeeId=134&page=undefined&limit=10`,
            responseCode: 200,
            data: {
                ratingListingCount: 1,
                ratingListing: [
                    {
                        reviewCycleId: 150,
                        id: 134,
                        employeeId: 'SR001',
                        firstName: 'Moly',
                        lastName: 'Agarwal',
                        checkInRating: 3.0
                    }
                ]
            }
        },
        newReviewCycleData: {
            url: `${BASE_URL}/api/analytics/rating-listing/?organisationId=1&reviewCycleId=150&ratingType=meetsExpectations&employeeId=-99&page=1&limit=10`,
            responseCode: 200,
            data: {
                ratingListingCount: 1,
                ratingListing: [
                    {
                        reviewCycleId: 155,
                        id: 134,
                        employeeId: 'SR0090',
                        firstName: 'Moly',
                        lastName: 'Agarwal',
                        checkInRating: 3.0
                    }
                ]
            }
        },
        paginationData: {
            url: `${BASE_URL}/api/analytics/rating-listing/?organisationId=1&reviewCycleId=150&ratingType=meetsExpectations&employeeId=-99&page=2&limit=10`,
            responseCode: 200,
            data: {
                ratingListingCount: 1,
                ratingListing: [
                    {
                        reviewCycleId: 150,
                        id: 110,
                        employeeId: `SR00100`,
                        firstName: 'Moly ',
                        lastName: 'Agarwal',
                        checkInRating: 3.0
                    }
                ]
            }
        }
    },

    LocationMockData: {
        pathname: '/reports/analytics/Ratings',
        state: {
            title: 'meetsExpectations',
            reviewCycle: 150,
            ratingLabel: 'Meets Expectations'
        },
        search: '',
        hash: '',
        key: ''
    },
    someLocationMockData: {
        pathname: '/reports/analytics/Ratings',
        state: {
            title: 'meetsExpectations',
            ratingLabel: 'Meets Expectations'
        },
        search: '',
        hash: '',
        key: ''
    }
};
