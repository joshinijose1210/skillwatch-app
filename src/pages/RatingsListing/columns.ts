import { TableColumnConfig } from '@medly-components/core';

export const ratingListingCols: TableColumnConfig[] = [
    {
        title: 'Review Cycle',
        field: 'reviewCycle'
    },
    {
        title: 'Employee Name (Employee ID)',
        field: 'employeeDetails'
    },
    {
        title: 'Check-in Rating',
        field: 'rating'
    }
];
