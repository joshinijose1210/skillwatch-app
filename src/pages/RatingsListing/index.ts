export { RatingsListing as default } from './RatingsListing';
