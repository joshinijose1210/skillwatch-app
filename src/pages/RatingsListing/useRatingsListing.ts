import { IReviewCycle } from '@pages/ReportsAnalytics/types';
import { useAppSelector } from '@slice';
import { useGetCyclesQuery, useGetEmployeeListQuery, useGetRatingListingQuery } from '@slice/services';
import { format } from 'date-fns';
import { useEffect, useMemo, useState } from 'react';
import { useLocation } from 'react-router-dom';
import { LocationState, RatingsData, ReviewCycleOptions } from './type';

export const useRatings = () => {
    const userDetails = useAppSelector(state => state.user);
    const { title = '', reviewCycle: reviewCycleId, ratingLabel } = useLocation().state as LocationState;
    const [activePage, setActivePage] = useState(1);
    const [reviewCycle, setReviewCycle] = useState(reviewCycleId ?? '');
    const [selectedEmployees, setSelectedEmployees] = useState([]);
    const [columnData, setColumnData] = useState([]);
    const [testPage, setTestPage] = useState<number>(); // for testing purpose

    const { data, isFetching: isLoading } = useGetRatingListingQuery(
        {
            path: '',
            params: [
                { name: 'organisationId', value: userDetails.organisationId },
                { name: 'reviewCycleId', value: reviewCycle },
                { name: 'ratingType', value: title ?? '' },
                { name: 'employeeId', value: selectedEmployees.join(',') || -99 },
                { name: 'page', value: activePage ?? testPage },
                { name: 'limit', value: 10 }
            ]
        },
        { refetchOnMountOrArgChange: true }
    );
    const { data: reviewCyclesData } = useGetCyclesQuery(
        {
            path: 'getAll',
            params: [{ name: 'organisationId', value: userDetails.organisationId }]
        },
        { refetchOnMountOrArgChange: true }
    );

    const { data: employees, isSuccess: isEmployeeListReceived } = useGetEmployeeListQuery(
        {
            path: '',
            params: [
                { name: 'teamId', value: -99 },
                { name: 'designationId', value: -99 },
                { name: 'organisationId', value: userDetails.organisationId },
                { name: 'roleId', value: -99 },
                { name: 'searchText', value: '' },
                { name: 'departmentId', value: -99 }
            ]
        },
        { refetchOnMountOrArgChange: true }
    );

    // const employeeList = useCallback(() => {
    //     let filteredList = [];
    //     if (isEmployeeListReceived) {
    //         filteredList = employees?.totalEmployees ? employees?.employees.filter((employee: any) => employee.status === true) : [];
    //     }
    //     let list = filteredList
    //         ? filteredList.map((employee: any) => {
    //               return {
    //                   value: employee.id,
    //                   label: `${employee.firstName} ${employee.lastName} (${employee.employeeId})`
    //               };
    //           })
    //         : [];
    //     list = list.filter((employee: any) => employee.value !== userDetails.id);

    //     list.sort((a: any, b: any) => a.label.localeCompare(b.label));
    //     return list;
    // }, [employees, isEmployeeListReceived, userDetails.id]);

    const reviewCycleList = useMemo(() => {
        return reviewCyclesData?.totalReviewCycles
            ? reviewCyclesData.reviewCycles.map((cycle: IReviewCycle) => {
                  if (!reviewCycleId && cycle.publish) setReviewCycle(cycle.reviewCycleId);
                  return {
                      value: cycle.reviewCycleId,
                      label: `${format(new Date(cycle.startDate), 'do MMM yyyy')} - ${format(new Date(cycle.endDate), 'do MMM yyyy')}`
                  };
              })
            : [];
    }, [reviewCycleId, reviewCyclesData?.reviewCycles, reviewCyclesData?.totalReviewCycles]);

    // const handleReviewCycle = useCallback((value: string) => {
    //     setReviewCycle(value);
    // }, []);

    // const handleSelectEmployee = (value: any) => {
    //     setSelectedEmployees(value);
    // };

    const handlePageChange = (page: number) => {
        setActivePage(page);
        setTestPage(page);
    };

    useEffect(() => {
        if (data?.ratingListing?.length) {
            const formattedData = data?.ratingListing.map((item: RatingsData) => ({
                reviewCycle:
                    reviewCycleList?.length &&
                    reviewCycleList?.find((cycle: ReviewCycleOptions) => cycle.value === item?.reviewCycleId).label,
                employeeDetails: `${item.firstName} ${item.lastName} (${item.employeeId})`,
                rating: item.checkInRating
            }));
            setColumnData(formattedData);
        } else {
            setColumnData([]);
        }
    }, [data?.ratingListing, reviewCycleList]);

    return {
        columnData,
        isLoading,
        activePage,
        handlePageChange,
        // handleReviewCycle,
        totalItems: data?.ratingListingCount || 0,
        reviewCycleList,
        reviewCycle,
        // employeeList,
        // handleSelectEmployee,
        selectedEmployees,
        ratingLabel
    };
};
