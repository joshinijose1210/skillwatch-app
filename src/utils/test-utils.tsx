import { ToastContainer } from '@medly-components/core';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import { store as rtkStore, setupStore } from '@slice';
import { RenderOptions, RenderResult, render } from '@testing-library/react';
import { defaultTheme } from '@theme';
import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import reduxMockStore from 'redux-mock-store';
import { ThemeProvider } from 'styled-components';
import { userMockState } from '../mocks/userMockState';
import { initialState } from '../slice';

interface CustomOptions extends RenderOptions {
    initialState?: any;
    initialEntries?: any;
}

interface WithThemeProviderProps {
    children?: React.ReactNode;
}

const mockStore = reduxMockStore(),
    store = mockStore({
        ...initialState,
        user: { ...initialState.user, groups: ['admin'] }
    });

const WithThemeProvider: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <ThemeProvider theme={defaultTheme}>
        <>
            <ToastContainer position="top-end" />
            {children}
        </>
    </ThemeProvider>
);

const WithStore: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <WithThemeProvider>{children}</WithThemeProvider>
    </Provider>
);

const WithRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <MemoryRouter>
        <WithThemeProvider>{children}</WithThemeProvider>
    </MemoryRouter>
);

const WithStoreAndRouter: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
    <Provider store={store}>
        <MemoryRouter>
            <WithThemeProvider>{children}</WithThemeProvider>
        </MemoryRouter>
    </Provider>
);

export const RTKRender = (ui: React.ReactElement, { initialState = userMockState, ...renderOptions }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);

    const RtkWrapper: React.FunctionComponent<WithThemeProviderProps> = ({ children }) => (
        <Provider store={store}>
            <MemoryRouter initialEntries={renderOptions.initialEntries}>
                <WithThemeProvider>{children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
    return { rtkStore, ...render(ui, { wrapper: RtkWrapper, ...renderOptions }) };
};
const customRender = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithThemeProvider, ...options });

export const renderWithStore = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStore, ...options });

export const renderWithRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithRouter, ...options });

export const renderWithStoreAndRouter = (ui: React.ReactElement, options?: RenderOptions): RenderResult =>
    render(ui, { wrapper: WithStoreAndRouter, ...options });

export const wrapper = (props: { children: React.ReactNode }, { initialState = userMockState }: CustomOptions = {}) => {
    const store = setupStore(initialState);
    setupListeners(store.dispatch);
    return (
        <Provider store={store}>
            <MemoryRouter>
                <WithThemeProvider>{props.children}</WithThemeProvider>
            </MemoryRouter>
        </Provider>
    );
};

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render, store };
