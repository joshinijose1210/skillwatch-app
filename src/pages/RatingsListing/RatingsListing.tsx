import { PageContent } from '@components';
import { CustomTable } from '@components/reusableComponents/CustomTable/CustomTable';
import ListHeader from '@components/reusableComponents/ListHeader';
import { StyledWrapper } from '@pages/EmployeeFeedback/EmployeeFeedback.styled';
import { FC, memo } from 'react';
import { ratingListingCols } from './columns';
import { useRatings } from './useRatingsListing';

export const RatingsListing: FC = memo(() => {
    const {
        columnData,
        isLoading,
        activePage,
        totalItems,
        handlePageChange,
        // handleReviewCycle,
        // reviewCycleList,
        // reviewCycle,
        // employeeList,
        // handleSelectEmployee,
        // selectedEmployees,
        ratingLabel
    } = useRatings();

    return (
        <PageContent>
            <ListHeader title={`Ratings ${ratingLabel && `(${ratingLabel})`}`} />
            <StyledWrapper>
                {/* <StyledMultiSelect
                    options={employeeList()}
                    variant="outlined"
                    placeholder="Select Employee Name"
                    label="Employee"
                    onChange={handleSelectEmployee}
                    values={selectedEmployees}
                    size="M"
                    minWidth="25rem"
                /> */}
                {/* <StyledSingleSelect
                    options={reviewCycleList}
                    label="Review Cycle"
                    placeholder="Select Review Cycle"
                    variant="outlined"
                    size="M"
                    onChange={handleReviewCycle}
                    value={reviewCycle}
                    minWidth="25rem"
                /> */}
            </StyledWrapper>
            <CustomTable
                columns={ratingListingCols}
                data={columnData}
                count={totalItems}
                isLoading={isLoading}
                activePage={activePage}
                handlePageChange={handlePageChange}
            />
        </PageContent>
    );
});

RatingsListing.displayName = 'RatingsListing';
