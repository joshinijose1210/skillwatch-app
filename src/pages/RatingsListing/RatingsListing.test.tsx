import { RTKRender, fireEvent, screen } from '@utils/test-utils';
import MockAdapter from 'axios-mock-adapter';
import { apiInstance } from '@utils';
import { RatingsListingMockData } from '@mocks/ratingsListing';
import { RatingsListing } from './RatingsListing';
import * as reactRouterDom from 'react-router-dom';

const mock = new MockAdapter(apiInstance, { onNoMatch: 'throwException' });

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useLocation: jest.fn()
}));

beforeAll(() => {
    const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');
    useLocationMock.mockReturnValue(RatingsListingMockData.LocationMockData);
});

describe('RatingsListing', () => {
    it('should render properly', () => {
        const { ratingListing, reviewCycle } = RatingsListingMockData.get;
        mock.onGet(ratingListing.url)
            .reply(ratingListing.responseCode, ratingListing.data)
            .onGet(reviewCycle.url)
            .reply(reviewCycle.responseCode, reviewCycle.data);
        const { container } = RTKRender(<RatingsListing />);
        expect(container).toMatchSnapshot();
    });

    // it('should test employee dropdown', async () => {
    //     const { searchEmployeeData, employeesData } = RatingsListingMockData.get;
    //     mock.onGet(employeesData.url)
    //         .reply(employeesData.responseCode, employeesData.data)
    //         .onGet(searchEmployeeData.url)
    //         .replyOnce(searchEmployeeData.responseCode, searchEmployeeData.data);
    //     RTKRender(<RatingsListing />);
    //     const empDropdown = screen.getByTestId('employee-dropdown');
    //     fireEvent.click(empDropdown);
    //     fireEvent.click(await screen.findByText('Moly Agarwal (SR001)'));
    //     const searchResult = screen.getAllByText('Moly Agarwal (SR001)');
    //     expect(searchResult.length).toEqual(2);
    // });

    it('should set reviewCycleId when its not passed from locationState and test pagination', async () => {
        const useLocationMock = jest.spyOn(reactRouterDom, 'useLocation');
        useLocationMock.mockReturnValue(RatingsListingMockData.someLocationMockData);
        const { ratingListing, reviewCycle, paginationData } = RatingsListingMockData.get;
        mock.onGet(reviewCycle.url)
            .replyOnce(reviewCycle.responseCode, reviewCycle.data)
            .onGet(ratingListing.urlWithPage)
            .replyOnce(ratingListing.responseCode, ratingListing.data)
            .onGet(paginationData.url)
            .replyOnce(paginationData.responseCode, paginationData.data);
        RTKRender(<RatingsListing />);
        const pagnBtn = await screen.findAllByText(2);
        fireEvent.click(pagnBtn[pagnBtn.length - 1]);
        expect((await screen.findAllByText('Moly Agarwal (SR00100)'))[0]).toBeInTheDocument();
    });

    // it('should test Review Cycle dropdown', async () => {
    //     const { ratingListing, newReviewCycleData } = RatingsListingMockData.get;
    //     mock.onGet(ratingListing.urlWithPage)
    //         .reply(ratingListing.responseCode, ratingListing.data)
    //         .onGet(newReviewCycleData.url)
    //         .reply(newReviewCycleData.responseCode, newReviewCycleData.data);
    //     RTKRender(<RatingsListing />);
    //     const reviewCycleDropdown = screen.getByTestId('review-cycle-dropdown');
    //     fireEvent.click(reviewCycleDropdown, { target: { value: 155 } });
    //     // expect((await screen.findAllByText('Moly Agarwal (SR0090)'))[0]).toBeInTheDocument();
    //     expect(true).toBeTruthy();
    // });
});
