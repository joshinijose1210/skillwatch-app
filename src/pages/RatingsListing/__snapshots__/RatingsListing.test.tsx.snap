// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingsListing should render properly 1`] = `
<div>
  <main
    class="sc-1ehavxz-0 dDyhBJ"
  >
    <div
      class="sc-h70dbj-0 fQmYvl row expanded space-between align-center marginBottom-2"
    >
      <div
        class="sc-w8e7z6-2"
      >
        <h3
          class="sc-1ntgl3l-0 kcLBOi sc-w8e7z6-0 ibnmt"
          textalign="initial"
          textcolor="inherit"
          textvariant="h3"
          textweight="Medium"
        >
          Ratings (Meets Expectations)
        </h3>
      </div>
    </div>
    <div
      class="sc-18u9mah-0 bePzMh"
    />
    <div
      class="sc-8tbhpn-3"
    >
      <div
        class="sc-1msmxxr-0 iWEzjW"
      />
      <table
        actions=""
        class="sc-1msmxxr-1 jwFsPF sc-8tbhpn-2 kCDmJI"
        columns="[object Object],[object Object],[object Object]"
        defaultactivepage="1"
        defaultsortorder="desc"
        itemsperpage="10"
        rowclickdisablekey=""
        tabindex="-1"
        totalitems="0"
      >
        <thead
          class="sc-uxzsxp-0 haNltq"
        >
          <tr
            class="sc-1jr91bg-0 hJzMcO"
            gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
          >
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Review Cycle
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Employee Name (Employee ID)
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
            <th
              class="sc-hjmm4m-0 iOyNiA"
              tablesize="M"
            >
              <button
                class="sc-hjmm4m-2 kpjakb"
                tablesize="M"
              >
                <span
                  class="sc-1ntgl3l-0 ghpJwn"
                  textalign="initial"
                  textcolor="inherit"
                >
                  Check-in Rating
                </span>
              </button>
              <span
                class="sc-hjmm4m-1 cuEDpa"
              />
            </th>
          </tr>
        </thead>
        <tbody
          class="sc-c7cniy-0 kShdfA"
        >
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
            tabindex="0"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
            tabindex="1"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
            tabindex="2"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
            tabindex="3"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
          <tr
            class="sc-1yxujaa-0 bgyIwA"
            gridtemplatecolumns=" minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr)"
            style="grid-template-columns: minmax(100px, 1fr) minmax(100px, 1fr) minmax(100px, 1fr);"
            tabindex="4"
          >
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
            <td
              class="sc-1e51eaa-0 hFOqbY"
              tablesize="M"
            >
              <div
                class="sc-15vbyef-0 bHPECM"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </main>
</div>
`;
